packages/babel-plugin-transform-flow-comments/src/index.js

```diff
@@ -135,14 +135,31 @@ export default declare(api => {
 
       Class(path) {
         const { node } = path;
-        if (node.typeParameters) {
-          const typeParameters = path.get("typeParameters");
+        if (node.typeParameters || node.implements) {
+          const comments = [];
+          if (node.typeParameters) {
+            const typeParameters = path.get("typeParameters");
+            comments.push(
+              generateComment(typeParameters, typeParameters.node).replace(
+                /^:: /,
+                "",
+              ),
+            );
+            typeParameters.remove();
+          }
+          if (node.implements) {
+            const impls = path.get("implements");
+            comments.push(
+              "implements " +
+                impls
+                  .map(impl => generateComment(impl).replace(/^:: /, ""))
+                  .join(", "),
+            );
+            delete node["implements"];
```

Comment:
Doesn't `impls.remove()` work? It's better not to mutate the ast directly, since `path` methods sometimes do some clean up when mutating nodes.

Reply:
`impls` is an array, there's no `.remove()`
should I clone the node, modify and `path.replaceWith(...)` with the clone?

Reply:
Oh you can `impls.forEach(p => p.remove())`

Reply:
I remembered I tried it before, i ended up with `class Foo implements {`

Reply:
It's ok like this then.